plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -201,11 +257,15 @@ collectRecSelsRule recorder = define (cmapWithPrio LogShake recorder) $
                 -- the OverloadedRecordDot pragma
                 exts = getEnabledExtensions tmr
                 recSels = mapMaybe (rewriteRange pm) (getRecordSelectors tmr)
+            uniques <- liftIO $ replicateM (length recSels) (hashUnique <$> newUnique)
```

Comment:
Is `newUnique` threadsafe? We run handlers in individual threads, so there could well be multiple instances of this running at once.

Reply:
Yes https://stackoverflow.com/questions/70515566/when-is-data-unique-thread-safe

Reply:
we might have to worry about contention on the ioref if we end up using this everywhere, but I expect that won't manifest until we're using it a lot. We should remember to try and check when we're nearly done